### PR TITLE
Separate poller response envelopes

### PR DESCRIPTION
**Pull request by catalinaperalta**
This PR creates the PollerResponse envelopes that contain poller related functionality (PollUntilDone and the initialized Poller). Also separating pollers from widget response envelopes. 

Headers are excluded for LRO operations. 
Pager related improvements and specific response envelopes will be added in the LRO+pager PR.

### CODE REVIEW

File: src/transform/transform.ts [hunk 1]
```diff
@@ -564,43 +563,51 @@ function createResponseType(codeModel: CodeModel, group: OperationGroup, op: Ope
...
-        }
       }
     }
     // create pageable type info
     if (isPageableOperation(op)) {
       if (codeModel.language.go!.pageableTypes === undefined) {
         codeModel.language.go!.pageableTypes = new Array<PagerInfo>();
       }
-      const name = `${(<SchemaResponse>response).schema.language.go!.name}Pager`;
+      let suffix = 'Pager';
```
Comment (jhendrixMSFT) [inline src/transform/transform.ts:609]: Should be `const`, or you could just do `const name = <name>+'Pager;'`.

File: src/generator/operations.ts [hunk 2]
```diff
@@ -1039,14 +1043,19 @@ function generateReturnsInfo(op: Operation, forHandler: boolean): string[] {
...
-    if (!forHandler && isPageableOperation(op)) {
+    const firstResp = <SchemaResponse>op.responses![0];
+    // must check pageable first as all pageable operations are also schema responses, 
+    // but LRO operations that return a pager are an exception and need to return LRO specific
+    // responses
+    if (!forHandler && isPageableOperation(op) && !isLROOperation(op)) {
       returnType = op.language.go!.pageableType.name;
     } else if (isSchemaResponse(firstResp)) {
       returnType = '*' + firstResp.schema.language.go!.responseType.name;
+      if (isLROOperation(op) || (isLROOperation(op) && forHandler)) {
```
Comment (jhendrixMSFT) [inline src/generator/operations.ts:1054]: The second half of the OR clause will never be true.

Comment (catalinaperalta) [reply src/generator/operations.ts:1054]: Oops, I thought I had removed this. Will do

File: src/transform/transform.ts (hunk 1 above)
Comment (catalinaperalta) [reply src/transform/transform.ts:593]: Done

File: src/generator/operations.ts (hunk 2 above)
Comment (catalinaperalta) [reply src/generator/operations.ts]: Done

**Pull request merged**